test/e2e/lazyload_test.go

```diff
@@ -0,0 +1,320 @@
+	}
+	ginkgo.By("slimemodule lazyload installs successfully")
+}
+
+func createExampleApps(f *framework.Framework) {
+	cs := f.ClientSet
+
+	exampleAppsYaml := readFile(test, "config/bookinfo.yaml")
+	framework.RunKubectlOrDieInput(nsApps, exampleAppsYaml, "create", "-f", "-")
+	defer func() {
+		testResourceToDelete = append(testResourceToDelete, &TestResource{Namespace: nsApps, Contents: exampleAppsYaml})
+	}()
+
+	// check
+	for i := 0; i < 60; i++ {
+		pods, err := cs.CoreV1().Pods(nsApps).List(context.TODO(), metav1.ListOptions{})
+		framework.ExpectNoError(err)
+		if len(pods.Items) != 6 {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+		for _, pod := range pods.Items {
+			err = e2epod.WaitForPodRunningInNamespace(cs, &pod)
+			framework.ExpectNoError(err)
+		}
+		break
+	}
+	ginkgo.By("example apps install successfully")
+}
+
+func createServiceFence(f *framework.Framework) {
+	svfGroup := "microservice.slime.io"
+	svfVersion := "v1alpha1"
+	svfResource := "servicefences"
+	svfName := "productpage"
+
+	sidecarGroup := "networking.istio.io"
+	sidecarVersion := "v1beta1"
+	sidecarResource := "sidecars"
+	sidecarName := "productpage"
+
+	// create CR ServiceFence
+	serviceFenceYaml := readFile(test, "samples/lazyload/servicefence_productpage.yaml")
+	framework.RunKubectlOrDieInput(nsApps, serviceFenceYaml, "create", "-f", "-")
+	defer func() {
+		testResourceToDelete = append(testResourceToDelete, &TestResource{Namespace: nsApps, Contents: serviceFenceYaml})
+	}()
+
+	// check
+	svfGvr := schema.GroupVersionResource{
+		Group:    svfGroup,
+		Version:  svfVersion,
+		Resource: svfResource,
+	}
+
+	svfCreated := false
+	for i := 0; i < 60; i++ {
+		_, err := f.DynamicClient.Resource(svfGvr).Namespace(nsApps).Get(context.TODO(), svfName, metav1.GetOptions{})
+		if err != nil {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+		svfCreated = true
+		break
+	}
+	if svfCreated != true {
+		framework.Failf("Failed to create servicefence.\n")
+	}
+
+	sidecarGvr := schema.GroupVersionResource{
+		Group:    sidecarGroup,
+		Version:  sidecarVersion,
+		Resource: sidecarResource,
+	}
+
+	sidecarCreated := false
+	for i := 0; i < 60; i++ {
+		_, err := f.DynamicClient.Resource(sidecarGvr).Namespace(nsApps).Get(context.TODO(), sidecarName, metav1.GetOptions{})
+		if err != nil {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+		sidecarCreated = true
+		break
+	}
+	if sidecarCreated != true {
+		framework.Failf("Failed to create sidecar.\n")
+	}
+
+	ginkgo.By("serviceFence and Sidecar create successfully")
+}
+
+func updateSidecar(f *framework.Framework) {
+	sidecarGroup := "networking.istio.io"
+	sidecarVersion := "v1beta1"
+	sidecarResource := "sidecars"
+	sidecarName := "productpage"
+
+	pods, err := f.ClientSet.CoreV1().Pods(nsApps).List(context.TODO(), metav1.ListOptions{})
+	framework.ExpectNoError(err)
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "ratings") {
```

Comment:
curl from every ratings pods?

Reply:
we'd better send a single req and confirm that it's sent to global sidecar at that point.

Reply:
Ok，I will change this part in the future.